elixir/apps/web/lib/web/components/form_components.ex

```diff
@@ -358,6 +358,85 @@ defmodule Web.FormComponents do
     """
   end
 
+  ### Dialogs ###
+
+  attr :id, :string, required: true, doc: "The id of the dialog"
+  attr :class, :string, default: "", doc: "Custom classes to be added to the button"
+  attr :style, :string, default: "danger", doc: "The style of the button"
+  attr :icon, :string, default: "hero-trash-solid", doc: "The icon of the button"
+  attr :size, :string, default: "md", doc: "The size of the button"
+  attr :on_confirm, :string, required: true, doc: "The phx event to broadcast on confirm"
+
+  attr :on_confirm_id, :string,
+    default: nil,
+    doc: "The phx event id value to broadcast on confirm"
+
+  slot :dialog_title, doc: "The title of the dialog"
+  slot :dialog_content, doc: "The content of the dialog"
+  slot :dialog_confirm_button, doc: "The content of the confirm button of the dialog"
+  slot :dialog_cancel_button, doc: "The content of the cancel button of the dialog"
+  slot :inner_block, required: true, doc: "The label for the button"
+
+  def button_with_confirmation(assigns) do
+    ~H"""
+    <dialog
+      id={"#{@id}_dialog"}
+      class={[
+        "backdrop:bg-slate-800/75 bg-transparent",
+        "p-4 w-full max-w-2xl md:inset-0 max-h-full",
+        "overflow-y-auto overflow-x-hidden",
+        "justify-center items-center"
+      ]}
+    >
+      <form method="dialog">
+        <div class="relative bg-white rounded-lg shadow">
```

Comment:
Closes on click outside 

```suggestion
        <div
          phx-click-away={JS.dispatch("click", to: "[data-dialog-action='cancel']")}
          class="relative bg-white rounded-lg shadow"
        >

```